### PR TITLE
Sample menu for screenshare, devices and mutes

### PR DESCRIPTION
**Pull request by kubaau**

### CODE REVIEW

**Comment by kubaau** [comment]
I'll explain this later XD

File: DolbyIO/Source/Private/DolbyIOSubsystem.cpp [hunk 1]
```diff
@@ -281,6 +281,10 @@ void UDolbyIOSubsystem::UpdateStatus(conference_status Status)
 		case conference_status::left:
 		case conference_status::error:
 			BroadcastEvent(OnDisconnected);
+			{
+				FScopeLock Lock{&RemoteParticipantsLock};
+				RemoteParticipants.Reset();
```
Comment (kubaau) [inline DolbyIO/Source/Private/DolbyIOSubsystem.cpp:286]: Why Reset and not Empty? Reset keeps allocations which may be unncessary.

File: DolbyIO/Source/Private/DolbyIOSubsystem.cpp [hunk 2]
```diff
@@ -310,6 +314,10 @@ void UDolbyIOSubsystem::Connect(const FString& ConferenceName, const FString& Us
 	UserInfo.name = ToStdString(UserName);
 	UserInfo.externalId = ToStdString(ExternalID);
 	UserInfo.avatarUrl = ToStdString(AvatarURL);
+	{
+		FScopeLock Lock{&RemoteParticipantsLock};
+		RemoteParticipants.Reset();
```
Comment (kubaau) [inline DolbyIO/Source/Private/DolbyIOSubsystem.cpp:319]: This should also be in DemoConference. It probably even deserves its own method at that point.

**Comment by kubaau** [comment]
I can't approve since this PR is mine ;)

File: DolbyIO/Source/Private/DolbyIOSubsystem.cpp [hunk 3]
```diff
@@ -116,6 +116,10 @@ void UDolbyIOSubsystem::Initialize(const FString& Token)
 			            const FDolbyIOParticipantInfo Info = ToFDolbyIOParticipantInfo(Event.participant);
 			            DLB_UE_LOG("Participant status added: UserID=%s Name=%s ExternalID=%s Status=%s", *Info.UserID,
 			                       *Info.Name, *Info.ExternalID, *ToString(*Event.participant.status));
+			            {
+				            FScopeLock Lock{&RemoteParticipantsLock};
+				            RemoteParticipants.Emplace(*Info.UserID, Info);
```
[the remark this answers is not in the document]
Comment (Adam1-3) [reply DolbyIO/Source/Private/DolbyIOSubsystem.cpp:121]: with emptying the collection it is safe enough ;)

**Review by Adam1-3 — DISMISSED** [review]

**Review by Adam1-3 — APPROVED** [review]

**Pull request merged**